fmp_uri.go

```diff
@@ -0,0 +1,67 @@
+package rpc
+
+import (
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"net"
+	"net/url"
+)
+
+type fmpURIScheme string
+
+const (
+	fmpSchemeStandard fmpURIScheme = "fmprpc"
+	fmpSchemeTLS      fmpURIScheme = "fmprpc+tls"
+)
+
+type FMPURI struct {
+	Scheme   fmpURIScheme
+	HostPort string
+	Host     string
+}
+
+var errInvalidFMPScheme = errors.New("invalid framed msgpack rpc scheme")
+var errNoHost = errors.New("missing host in framed msgpack rpc URI")
+
+func ParseFMPURI(s string) (*FMPURI, error) {
```

Comment:
Please add some godoc comments so we can slowly move towards `govet` passing on this repo.